### ISSUE TITLE
[Block] File: PDF preview display in editor after toggling inline embed option

### ISSUE THREAD

**Issue by ironprogrammer** [WordPress/gutenberg#62120]
### Description

When the "Show inline embed" option is toggled off and back on for a PDF (using the File block), the PDF is displayed as a gray box in the editor (backend). Occurs in Safari, but not Chrome or Firefox.

This issue was also mentioned in https://github.com/WordPress/gutenberg/issues/55493#issuecomment-1778829000, though that issue and [resulting PR](https://github.com/WordPress/gutenberg/pull/55667) focused on the frontend display of inline PDFs, not display in the editor.

### Step-by-step reproduction instructions

1. Using the File block, upload or insert a PDF file. Observe that it displays in the editor as expected.
2. Toggle OFF the "Show inline embed" option. The PDF preview will be hidden.
3. Toggle ON the "Show inline embed" option. Observe that the PDF preview displays as a gray box.

Refreshing the page will restore the PDF preview if the option is enabled.

### Screenshots, screen recording, code snippet

Animated GIF that shows:
- ✅ Frontend PDF embed functions as expected (see https://github.com/WordPress/gutenberg/issues/55493).
- ❌ Backend/editor PDF embed display is broken after toggling the inline embed option.

![pdf-embed-retest](https://github.com/WordPress/gutenberg/assets/824344/48201cf8-3b85-46a1-ac1f-6328462713c9)

### Environment info

- macOS 14.5, Safari 17.5
- WordPress 6.5.3 and trunk
- Gutenberg 18.4.1 (tested both active and inactive)

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by talldan** [comment]
I also saw this issue here was reported, which seems similar - https://github.com/WordPress/gutenberg/issues/62037

cc @DAreRodz

**Comment by ironprogrammer** [comment]
In response to @talldan:
> I also saw this issue here was reported, which seems similar - #62037

Yeah, I saw that as well, though the description seems to be a dupe of #55493 (frontend focused), which I was unsure should be marked as such and the original reopened, as there haven't been additional reproduction confirmations as of yet.

**Comment by DAreRodz** [comment]
Yup, it seems to be a different bug this time. The previous one specifically happened in the front-end part, whereas this one is related to the editor. Would the fix be similar, though? 🤔

**Comment by t-hamano** [comment]
I'm a Windows user so I can't test in Safari, but if this issue is related to caching, but I'd like to know if either of the following approaches would work:


Add key prop:

```javascript
<object
	key={ Date.now() }
	className="wp-block-file__preview"
	data={ href }
	type="application/pdf"
	aria-label={ __(
		'Embed of the selected PDF file.'
	) }
/>
```

Add get parameter to href:

```javascript
<object
	key={ Date.now() }
	className="wp-block-file__preview"
	data={ href + '?v=' + Date.now() }
	type="application/pdf"
	aria-label={ __(
		'Embed of the selected PDF file.'
	) }
/>
```


cc @dhruvikpatel18

**Comment by dhruvikpatel18** [comment]
@t-hamano Thanks for the confirming, Yes second approach you have shared would work.
